File: src/Students.Domain/Students/Student.cs
using Students.Domain.Enums;
namespace Students.Domain.Students;

public class Student
{
    public Guid Id { get; }
    public String FullName { get; }
    public Gender Gender { get;}
    public DateTime BirthDate { get; }
    public Single AverageGrade { get;}
    public String[] SpecialMarks { get;}
    public Guid StudentGroupId { get;}
    public Boolean HasScholarship => AverageGrade >= 4;

    public Student(Guid id,String fullName, Gender gender, DateTime birthDate, Single averageGrade, string[] specialMarks, Guid studentGroupId)
    {
        Id = id;
        FullName = fullName;
        Gender = gender;
        BirthDate = birthDate;
        AverageGrade = averageGrade;
        SpecialMarks = specialMarks;
        StudentGroupId = studentGroupId;
    }

    public Int32 GetAge(DateTime currentDate)
    {
        Int32 age = currentDate.Year - BirthDate.Year;

        if (currentDate < BirthDate.AddYears(age))
            age--;

        return age;
    }
}

File: src/Students.BackOffice/client-app/src/domain/Students/Student.ts
import { Page } from '../../tools/types/page';
import { Gender } from './Gender';

export class Student {
  
	constructor(
		public readonly id: string,
		public readonly fullName: string,
		public readonly gender: Gender,
		public readonly age: number,
		public readonly averageGrade: number,
		public readonly specialMarks: string[],
		public readonly studentGroupId: string
	) { }
	public get hasScholarShip(): boolean{
		return this.averageGrade >= 4;
	}
	public get specialMarksText(): string {
		return this.specialMarks.join(', ')
	}
}

export function mapToStudentsPage(data: any): Page<Student> {
	return Page.convert(data, mapToStudent);
}

export function mapToStudents(data: any[]): Student[] {
	return data.map(mapToStudent);
}

export function mapToStudent(data: any): Student {
	return new Student(
		data.id,
		data.fullName,
		data.gender,
		data.age,
		data.averageGrade,
		data.specialMarks,
		data.studentGroupId
	);
}
